pkg/build/frontend.go

```diff
@@ -165,7 +165,7 @@ func resolveStates(ctx context.Context, bopts *BOpts, platform ocispecs.Platform
 			resolverOpts.ImageOpt = &sourceresolver.ResolveImageOpt{
 				ResolveMode: llb.ResolveModePreferLocal.String(),
 			}
-			_, digest, img, err := bopts.Resolver.ResolveImageConfig(ctx, ref.String(), resolverOpts)
+			_, digest, img, err := bopts.Resolver.ResolveImageConfig(ctx, resolvedBaseStageName.Result, resolverOpts)
```

Comment:
Can we add a comment to `resolveStates` that details why we don't use the ref here and how the constructed `pname` value is used?